### PR TITLE
feat(upgrade): support LTS release channel

### PR DESCRIPTION
**Pull request by dsherret**

### CODE REVIEW

File: cli/tools/upgrade.rs [hunk 1]
```diff
@@ -330,8 +330,15 @@ pub fn check_for_upgrades(
...
       }
-      // TODO(bartlomieju)
-      ReleaseChannel::Lts => unreachable!(),
+      ReleaseChannel::Lts => {
+        log::info!(
+          "{} {} → {} {}",
+          colors::green("A new LTS release of Deno is available:"),
+          colors::cyan(version::DENO_VERSION_INFO.deno),
+          colors::cyan(&upgrade_version),
+          colors::italic_gray("Run `deno upgrade` to install it.")
```
Comment (bartlomieju) [inline cli/tools/upgrade.rs:339]:
```suggestion
          colors::italic_gray("Run `deno upgrade lts` to install it.")
```

File: cli/tools/upgrade.rs [hunk 2]
```diff
@@ -623,65 +627,30 @@ fn select_specific_version_for_upgrade(
...
-        );
-        return Ok(None);
-      }
-
-      Ok(Some(AvailableVersion {
-        version_or_hash: version,
-        release_channel,
-      }))
+      version::DENO_VERSION_INFO.release_channel == ReleaseChannel::Rc
+        && version::DENO_VERSION_INFO.deno == version
```
Comment (bartlomieju) [inline cli/tools/upgrade.rs:638]: Now it occurred to me that we can use the same check for all release channels, but canary - we want to check if the current binary's release channel is the one requested and then compare if `version::DENO_VERSION_INFO.deno` is the same

Comment (dsherret) [reply cli/tools/upgrade.rs]: Done.

**Review by bartlomieju — COMMENTED** [review]
Note to self, when no LTS is available the error message is bad:

```
Current Deno version: v2.0.0-rc.1
Looking up long term support version
error: Not found.
```

**Review by bartlomieju — APPROVED** [review]
LGTM!

**Pull request merged**